### ISSUE TITLE
Multi-task self-supervised learning for Robust Speech Recognition

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#19]
## [Multi-task self-supervised learning for Robust Speech Recognition](https://arxiv.org/abs/2001.09239)
*Despite the growing interest in unsupervised learning, extracting meaningful knowledge from unlabelled audio remains an open challenge. To take a step in this direction, we recently proposed a problem-agnostic speech encoder (PASE), that combines a convolutional encoder followed by multiple neural networks, called workers, tasked to solve self-supervised problems (i.e., ones that do not require manual annotations as ground truth). PASE was shown to capture relevant speech information, including speaker voice-print and phonemes. This paper proposes PASE+, an improved version of PASE for robust speech recognition in noisy and reverberant environments. To this end, we employ an online speech distortion module, that contaminates the input signals with a variety of random disturbances. We then propose a revised encoder that better learns short- and long-term speech dynamics with an efficient combination of recurrent and convolutional networks. Finally, we refine the set of workers used in self-supervision to encourage better cooperation.
Results on TIMIT, DIRHA and CHiME-5 show that PASE+ significantly outperforms both the previous version of PASE as well as common acoustic features. Interestingly, PASE+ learns transferable representations suitable for highly mismatched acoustic conditions.*

### Cite this paper
```bibtex
@misc{ravanelli2020multitask,
    title={Multi-task self-supervised learning for Robust Speech Recognition},
    author={Mirco Ravanelli and Jianyuan Zhong and Santiago Pascual and Pawel Swietojanski and Joao Monteiro and Jan Trmal and Yoshua Bengio},
    year={2020},
    eprint={2001.09239},
    archivePrefix={arXiv},
    primaryClass={eess.AS}
}
```

**Comment by juanmc2005** [comment]
### Intro
This paper introduces an improvement on the previously presented model PASE. The new PASE+ uses noise disturbances to learn more robust representations, and adds more "worker" neural networks to learn different self-supervising tasks.
The authors hypothesize that these workers can learn different "views" of the data, thus encouraging generalized representations.

### Architecture
![image](https://user-images.githubusercontent.com/5760780/74641196-30cb3700-5171-11ea-9539-f796f5e1a4ba.png)
The model starts with a speech distortion module that adds different types of noise to the waveform, each with its own probability. This new signal is then fed to the main encoder, which is composed of a SincNet, 7 convolutional layers and a QRNN.
Finally, the workers are feed forward neural networks with 256 hidden units.
**Regression workers** optimize the MSE with respect to known transformations, like MFCC or LPS (Low Power Spectrum). In PASE+, they add workers for Gammatone and FBANKS, they estimate longer contexts (output) and from longer windows (input).
**Binary tasks** include LIM (Local Info Max) and GIM (Global Info Max). These tasks sample positive and negative embeddings with respect to the anchor and optimize a triplet loss. GIM differs from LIM in that the positive examples are obtained by averaging over the entire "sentence" of the speaker.

Finally, the encoder and all workers are jointly trained with backpropagation, using the average of all worker losses as the loss. They have found that weighting specific workers while training results in lower overall performance.

### Results
They perform the self-supervised training on 50 hours of Librispeech, and they evaluate the model on TIMIT, DIRHA and CHiME-5.
The ablation studies on TIMIT show that the new elements of PASE+ are in fact contributing to better representing the speech signal.
Furthermore, they find that PASE+ features outperforms hand-crafted features on TIMIT, DIRHA and CHiME-5. They also found that MFCCs can be complementary to PASE+ features in some cases, leading to even better performance.
